htdocs/core/lib/images.lib.php

```diff
@@ -306,35 +306,94 @@ function dol_imageResizeOrCrop($file, $mode, $newWidth, $newHeight, $src_x = 0,
-    if ($exif === false) {
-        return false;
-    }
-    $orientation = intval(@$exif['Orientation']);
-    if (!in_array($orientation, array(3, 6, 8))) {
-        return false;
-    }
-    $image = @imagecreatefromjpeg($file_path);
-    switch ($orientation) {
-        case 3:
-            $image = @imagerotate($image, 180, 0);
-            break;
-        case 6:
-            $image = @imagerotate($image, 270, 0);
-            break;
-        case 8:
-            $image = @imagerotate($image, 90, 0);
-            break;
-        default:
-            return false;
-    }
-    $success = imagejpeg($image, $file_path);
-    // Free up memory (imagedestroy does not delete files):
-    @imagedestroy($image);
-    return $success;
+	return correctExifImageOrientation($file_path, $file_path);
 }
 
 
+/**
+ * Add exif orientation correction for image
+ *
+ * @param string $fileSource Full path to source image to rotate
+ * @param string $fileDest string : Full path to image to rotate | false return gd img  | null  the raw image stream will be outputted directly
+ * @param int $quality output image quality
+ * @return bool : true on success or false on failure or gd img if $fileDest is false.
+ */
+function correctExifImageOrientation($fileSource, $fileDest, $quality = 95)
+{
+	if (function_exists('exif_read_data') ) {
+		$exif = exif_read_data($fileSource);
+		if($exif && isset($exif['Orientation'])) {
+
+			$infoImg = getimagesize($fileSource); // Get image infos
+
+			$orientation = $exif['Orientation'];
+			if($orientation != 1){
+				$img = imagecreatefromjpeg($fileSource);
+				$deg = 0;
+				switch ($orientation) {
+					case 3:
+						$deg = 180;
+						break;
+					case 6:
+						$deg = 270;
+						break;
+					case 8:
+						$deg = 90;
+						break;
+				}
+				if ($deg) {
+					if($infoImg[2] === 'IMAGETYPE_PNG') // In fact there is no exif on PNG but just in case
+					{
+						imagealphablending($img, false);
+						imagesavealpha($img, true);
+						$img = imagerotate($img, $deg, imageColorAllocateAlpha($img, 0, 0, 0, 127));
+						imagealphablending($img, false);
+						imagesavealpha($img, true);
+					}
+					else{
+						$img = imagerotate($img, $deg, 0);
+					}
+				}
+				// then rewrite the rotated image back to the disk as $fileDest
+				if($fileDest === false){
+					return $img;
+				}
+				else
+				{
+					// In fact there exif is only for JPG but just in case
+					// Create image on disk
+					$image = false;
+
+					switch($infoImg[2])
+					{
+						case IMAGETYPE_GIF:	    // 1
+							$image = imagegif($img, $fileDest);
+							break;
+						case IMAGETYPE_JPEG:    // 2
+							return imagejpeg($img, $fileDest, $quality);
+							break;
+						case IMAGETYPE_PNG:	    // 3
+							$image = imagepng($img, $fileDest, $quality);
+							break;
+						case IMAGETYPE_BMP:	    // 6
+							// Not supported by PHP GD
+							break;
+						case IMAGETYPE_WBMP:    // 15
+							$image = image2wbmp($img, $fileDest);
```

Comment:
This method is deprecated, we replaced with imagewbmp everywher in v11 so we should continue to not use it.

Reply:
@atm-john see https://github.com/Dolibarr/dolibarr/pull/12528